march_rqt_note_taker/src/march_rqt_note_taker/entry.py

```diff
@@ -1,28 +1,24 @@
-from python_qt_binding.QtCore import QDateTime, QObject, QTime
+from python_qt_binding.QtCore import QObject, QTime
+from rosgraph_msgs.msg import Log
 
 
 class Entry(QObject):
 
-    DEBUG = 1
-    INFO = 2
-    WARN = 3
-    ERROR = 4
-    FATAL = 5
-
-    LEVEL_LABELS = {
-        DEBUG: 'Debug',
-        INFO: 'Info',
-        WARN: 'Warn',
-        ERROR: 'Error',
-        FATAL: 'Fatal',
-    }
-
-    def __init__(self, content, level=INFO):
+    def __init__(self, content, is_error=False):
         super(Entry, self).__init__()
 
         self.content = content
         self.time = QTime.currentTime()
-        self.level = level
+        self.is_error = is_error
+
+    @classmethod
+    def from_ros_msg(cls, log_msg):
+        """Returns an Entry from a given ROS log message.
+
+        :type log_msg: rosgraph_msgs.msg.Log
+        :param log_msg: The message to convert
+        """
+        return cls(log_msg.msg, log_msg.level >= Log.ERROR)
```

Comment:
You can also use the time in the message header, to get a more precise time for the error.